pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -715,59 +715,33 @@ func (cb *ClusterBuilder) buildUpstreamClusterTLSContext(opts *buildClusterOpts,
 	case networking.ClientTLSSettings_DISABLE:
 		tlsContext = nil
 	case networking.ClientTLSSettings_ISTIO_MUTUAL:
-
 		tlsContext = &auth.UpstreamTlsContext{
 			CommonTlsContext: &auth.CommonTlsContext{},
 			Sni:              tls.Sni,
 		}
 
-		// For ISTIO_MUTUAL_TLS we serve certificates from two well known locations.
```

Comment:
This can not be removed. Because when people specify ISTIO_MUTUAL_TLS we should honour 
- Proxy Metadata certs
- default SDS location
and in mutual TLS we honour only DR certs.

Reply:
https://github.com/istio/istio/pull/26819 - This is where it was added. So we do not honour metadata certs in MUTUAL mode

Reply:
You just mean the comment?

Reply:
https://github.com/istio/api/blob/e3f75bd8b68620f99e0eec48dcdb06dc433285f7/networking/v1alpha3/destination_rule.proto#L914-L925

Reply:
Not just comment. The entire code. What that just means is for ISTIO_MUTUAL we do not honour certs provided in DR - For custom CA which specify certs via metadata they should be honoured. That is the change I made in that PR after discussing with @rshriram . So for 

MUTUAL - we only honour certs specified in DR as per the API contract. This will be removed soon when `PILOT_ALLOW_METADATA_CERTS_DR_MUTUAL_TLS` is removed.
ISTIO_MUTUAL - we honour default certs for standard deployments and metadata certs for custom CAs.

Reply:
I donot understand why you set Istio_MUTUAL mode when you want to use custom cert

Reply:
I think this is inconsistent with our API

Reply:
Because it was implemented like that from the beginning for MUTUAL i.e. to honour metadata certs and Shriram felt it is violating API contract for MUTUAL(as it should only honour certs in DR). So it was decided to change ISTIO_MUTUAL to honour those certs because as per API ISTIO_MUTUAL does not expect certs to be provided and remove that support form MUTUAL.

Why do you think it is inconsistent with ISTIO_MUTUAL API? and are you OK supporting metadata certs for MUTUAL? What is the difference?

Reply:
Can you look at the link above? 
	// Secure connections to the upstream using mutual TLS by presenting
	// client certificates for authentication.
	// Compared to Mutual mode, this mode uses certificates generated
	// automatically by Istio for mTLS authentication. When this mode is
	// used, all other fields in `ClientTLSSettings` should be empty.
	ClientTLSSettings_ISTIO_MUTUAL ClientTLSSettings_TLSmode = 3

Reply:
@howardjohn I am sure this behavior is not right

Reply:
I am not sure I understand the issue? is it "all other fields in ClientTLSSettings should be empty."?

In Rama's case, all other fields in `ClientTLSSettings` *are* empty. The DestinationRule just says "Use Istio mTLS". For custom cases, this means "read certs loaded into a pod, at a location specified in the proxy metadata".

I am not sure how moving to MUTUAL solves/supports this case. In typical case, ISTIO_MUTUAL DR isn't even really used and auto mtls would be?

Reply:
> For custom cases, this means "read certs loaded into a pod, at a location specified in the proxy metadata".

The hack point is we build ISTIO_MUTUAL ClientTLSSettings for it. We can build MUTUAL ClientTLSSettings for it.

Reply:
so its internal code organization? No user facing change?

Reply:
Yeah, I meant to.

Reply:
But we need to update the MUTUAL TLS docs and specify it can also load from proxy Metadata certs if available and they will be given preference over DR specified certs. That is how it is working now with MUTUAL TLS.

Reply:
Currently, the doc does not say anything about metadata certs.

Reply:
I know. But we wanted to update ISTIO_MUTUAL docs once we remove that flag. We can update autoMTLS docs with this.

Reply:
Before ISTIO_MUTUAL introduced, we only have `MUTUAL`, and ask users to specify the mounted secret path in the DR. This ISTIO_MUTUAL introduced and help to get rid of the internal TLS settings details. Allowing another config source to sepcify the details sounds like leaking API surface area and make it harder to change these settings in future. For example, do we know this would be compatible with BTS? IIRC, BTS might also consider to remove ISTIO_MUTUAL...

Just curious, are there any reason that the customized cert has to go through ISTIO_MUTUAL mode? Are we relying on some specific config settings there, such as ALPN stuff to get that use case working?

Reply:
I donot see any reason to make custom cert through ISTIO_MUTUAL.  That was just by experience.  And however violate ISTIO_MUTUAL.  I am not sure how could custom cert be compatible with istio signed certs when working at STRICT mode.